missioncontrol/api/views.py

```diff
@@ -83,39 +92,41 @@ def measure(request):
 
     if not all([channel_name, platform_name, measure_name, interval]):
         return HttpResponseBadRequest("All of channel, platform, measure, interval required")
-    data = cache.get(get_measure_cache_key(platform_name, channel_name, measure_name))
-    if not data:
+
+    datums = Datum.objects.filter(
+        series__build__channel__name=channel_name,
+        series__build__platform__name=platform_name,
+        series__measure__name=measure_name)
+
+    if not datums.exists():
         return HttpResponseNotFound("Data not available for this measure combination")
 
     # process min/max time based on parameters
     try:
         if start is not None:
-            min_time = datetime.datetime.fromtimestamp(int(start))
-            max_time = min_time + datetime.timedelta(seconds=int(interval))
+            min_time = datetime.datetime.fromtimestamp(int(start), tz=pytz.UTC)
+            datums = datums.filter(
+                timestamp__gt=min_time,
+                timestamp__lt=min_time + datetime.timedelta(seconds=int(interval)))
```

Comment:
I would imagine the start time would be inclusive (i.e. using `__gte`). Also curious if [range](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#range) would work here?

Reply:
Sure, let's use inclusive time. I didn't know about range! Let's use that too.